packages/faceted-search/src/components/Badges/BadgeMenu/BadgeMenuForm.component.js

```diff
@@ -119,6 +127,8 @@ const BadgeMenuForm = ({
 						type="submit"
 						label={t('APPLY', { defaultValue: 'Apply' })}
 						bsStyle="info"
+						disabled={rest.isLoading}
+						inProgress={rest.isLoading}
```

Comment:
IMO no need set `inProgress` to true when loading options, just disable submit button is enough. set `inProgress` to true when it's submitting.

Reply:
removed